### ISSUE TITLE
CA1801 Review Unused Parameters is raised during compilation with VB XML literals

### ISSUE THREAD

**Issue by adraut** [dotnet/roslyn-analyzers#1543]
#### Analyzer package
Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer
ReviewUnusedParametersAnalyzer

#### Repro steps
1. Create a VB full framework project.
2. Add the analyzers packages.
3. Add a class like the below to the project
```
Public Class C
    Public Function MakeString(val As String) As String
        Return <s><%= val %></s>.Value
    End Function
End Class1
```
4. Close the Class C file.
5. Build.
6. Observe that the diagnostic/warning `Parameter val of method MakeString is never used. Remove the parameter or use it in the method body` is in the build output and the Error List window in Visual Studio.
7. Open the Class C file.
8. Observe that the warning disappears from the Error List and that the diagnostic is no longer raised.

#### Expected behavior
The diagnostic is not raised during compilation or while the file is open.

#### Actual behavior
The diagnostic is raised during compilation, but not while the file is open.

**Comment by 333fred** [comment]
I noticed with a few other instances in the project-system repo just now, not just related to VB.

**Comment by mavasani** [comment]
> I noticed this with a few other instances in the project-system repo just now, not just related to VB.

@333fred Can we investigate this further and find if the root cause is IOperation bugs?

**Comment by dparvin** [comment]
I have been seeing this issue for a while in VS 2017 and later.  I have just started trying to use the analyzer version of code analysis and it is still happening.  I am of the opinion that the issue is with something the old FXCop and the new analyzers use in common.

**Comment by mavasani** [comment]
Seems related to https://github.com/dotnet/roslyn/issues/27564

**Comment by mavasani** [comment]
To clarify: this analyzer is based on IOperation, and we are possibly dropping lot of operations within VB XML literals, causing us to miss references.

**Comment by mavasani** [comment]
This issue is indeed due to https://github.com/dotnet/roslyn/issues/27564 - missing IOperation support for VB xml literals. Closing the issue as external.
